app/src/main/java/io/github/wulkanowy/ui/modules/account/AccountAdapter.kt

```diff
@@ -3,33 +3,71 @@ package io.github.wulkanowy.ui.modules.account
 import io.github.wulkanowy.databinding.ItemAccountBinding
 import io.github.wulkanowy.utils.getThemeAttrColor
 import javax.inject.Inject
 
-class AccountAdapter @Inject constructor() : RecyclerView.Adapter<AccountAdapter.ItemViewHolder>() {
+class AccountAdapter @Inject constructor() : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
-    var items = emptyList<Student>()
+    var items = emptyList<AccountItem<*>>()
 
     var onClickListener: (Student) -> Unit = {}
 
     override fun getItemCount() = items.size
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ItemViewHolder(
-        ItemAccountBinding.inflate(LayoutInflater.from(parent.context), parent, false)
-    )
+    override fun getItemViewType(position: Int) = items[position].viewType.id
 
-    @SuppressLint("SetTextI18n")
-    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
-        val student = items[position]
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+
+        return when (viewType) {
+            AccountItem.ViewType.HEADER.id -> HeaderViewHolder(HeaderAccountBinding.inflate(inflater, parent, false))
+            AccountItem.ViewType.ITEM.id -> ItemViewHolder(ItemAccountBinding.inflate(inflater, parent, false))
+            else -> throw IllegalStateException()
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is HeaderViewHolder -> bindHeaderViewHolder(holder.binding, items[position].value as Account)
+            is ItemViewHolder -> bindItemViewHolder(holder.binding, items[position].value as Student)
+        }
+    }
 
-        with(holder.binding) {
+    private fun bindHeaderViewHolder(binding: HeaderAccountBinding, account: Account) {
+        with(binding) {
+            accountHeaderEmail.text = account.email
+            accountHeaderType.setText(if (account.isParent) R.string.account_type_parent else R.string.account_type_student)
+        }
+    }
+
+    @SuppressLint("SetTextI18n")
+    private fun bindItemViewHolder(binding: ItemAccountBinding, student: Student) {
+        with(binding) {
             accountItemName.text = "${student.studentName} ${student.className}"
             accountItemSchool.text = student.schoolName
+            with(accountItemLoginMode) {
+                when (student.loginMode) {
+                    "API" -> {
+                        setText(R.string.account_login_mobile_api)
+                        visibility = VISIBLE
+                    }
+                    "HYBRID" -> {
+                        setText(R.string.account_login_hybrid)
+                        visibility = VISIBLE
+                    }
+                    else -> {
+                        visibility = GONE
+                    }
+                }
```

Comment:
Zamień loginMode na enuma

```suggestion
                visibility = when (Sdk.Mode.valueOf(student.loginMode)) {
                    Sdk.Mode.API -> {
                        setText(R.string.account_login_mobile_api)
                        VISIBLE
                    }
                    Sdk.Mode.HYBRID -> {
                        setText(R.string.account_login_hybrid)
                        VISIBLE
                    }
                    Sdk.Mode.SCRAPPER -> GONE
                }
```